app/Authority.php

```diff
@@ -122,6 +122,14 @@ public function sourceLinks()
         return $this->links()->where('type', 'source');
     }
 
+    public function scopeSwappedName($query, $name)
```

Comment:
I would suggest adding `where` to the scope name so that we know that it's a filter

```suggestion
    public function scopeWhereSwappedName($query, $name)
```

I'm also not sure if _swapped_ is the right name if you don't know what "unswapped" looks like.

```suggestion
    public function scopeWhereFirstNameLastName($query, $firstNameLastName)
```

Reply:
imho "scope" already suggest "where". also using `->whereSomehting()` might conflict with native magical `->whereAttribute('')` (I can't find it in documentation -> I mean this https://laracasts.com/discuss/channels/eloquent/whereattribute-function?page=1&replyId=402083 )

Reply:
> imho "scope" already suggest "where"

In the definition in does, but in the usage not necessarily. Compare e.g. 

```php
Authority::query()->swappedName($fullname)`
```
vs
```php
Authority::query()->whereSwappedName($fullname)`
```

scopes without a preposition suggest a trait to me, e.g. "Authorities that have a swapped name" rather than "Authorities where swapped name is XYZ".

I think overriding the `whereAttribute` is completely fine here, since we don't have such attribute anyway.